walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,166 @@
 
 package com.google.sps;
 
+import java.util.Arrays;
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.Set;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // Find events that match the requested number of attendees and duration
+    // and adds it to an ArrayList
+    Set<TimeRange> meetings = new HashSet<>();
+    long requiredDuration = request.getDuration();
+    Collection<String> requiredAttendees = request.getAttendees();
+    int requiredNumAttendees = requiredAttendees.size();
+    String[] requiredArr = new String[requiredNumAttendees];
+    String requiredPerson = "";
+    if (requiredNumAttendees >= 1) {
+      requiredAttendees.toArray(requiredArr);
+      requiredPerson = requiredArr[0];
+    }
+
+    // Convert the collection to an event array
+    Event[] evArr = events.toArray(new Event[events.size()]);
+    long timeLeft = TimeRange.WHOLE_DAY.duration();
+    
+    // Will be used in the for-each loop
+    int i = 0;
+    int k = 1;
+
+    for (Event event : events) {
+      Set<String> currAttendees = event.getAttendees();
+      int numAttendees = currAttendees.size();
+      String[] people = new String[numAttendees];
+      currAttendees.toArray(people);
+      String currPerson = "";
+
+      // Check how many attendees are in the current event
+      if (numAttendees == 1) {
+        currPerson = people[0];
+      }
+      else if (numAttendees > 1) {
+        currPerson = people[i];
```

Comment:
`i` is `0` at the start. I'm having trouble understanding why this `if`/`else if` is needed.